src/main/java/frc/robot/subsystems/vision/VisionHelper.java

```diff
@@ -0,0 +1,394 @@
+        minAreaRectCorners.add(new TargetCorner(minAreaRectCornersX[i], minAreaRectCornersY[i]));
+      }
+      Transform3d pose = getLoggedTransform3d(translation, rotation);
+      Transform3d altPose = getLoggedTransform3d(altTranslation, altRotation);
+      return (new PhotonTrackedTarget(
+          table.get("Tags/Yaw " + name, -1.0),
+          table.get("Tags/Pitch " + name, -1.0),
+          table.get("Tags/Area " + name, -1.0),
+          table.get("Tags/Skew " + name, -1.0),
+          table.get("Tags/Fiducial ID " + name, -1),
+          0, // These values are exclusively for object detection
+          0.0f, // These values are exclusively for object detection
+          pose,
+          altPose,
+          table.get("Tags/Pose Ambiguity " + name, -1),
+          minAreaRectCorners,
+          detectedCorners));
+    }
+
+    public static VisionConstants getLoggedVisionConstants(LogTable table) {
+      return new VisionConstants(
+          table.get("Vision Constants Name ", "Default"),
+          table.get("Vision Constants Transform ", new Transform3d()),
+          new Matrix<N3, N3>(
+              Nat.N3(),
+              Nat.N3(),
+              table.get("Vision Constants Intrinsics ", Matrix.eye(Nat.N3()).getData())),
+          new Matrix<N8, N1>(
+              Nat.N8(),
+              Nat.N1(),
+              table.get(
+                  "Vision Constants Distortion ",
+                  new double[] {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0})));
+    }
+  }
+
+  /**
+   * Updates the estimated position of the robot. Returns empty if:
+   *
+   * <ul>
+   *   <li>The timestamp of the provided pipeline result is the same as in the previous call to
+   *       {@code update()}.
+   *   <li>No targets were found in the pipeline results.
+   * </ul>
+   *
+   * @param cameraResult The latest pipeline result from the camera.
+   * @param cameraMatrix Camera calibration data that can be used in the case of no assigned
+   *     PhotonCamera.
+   * @param distCoeffs Camera calibration data that can be used in the case of no assigned
+   *     PhotonCamera
+   * @param strat The selected pose solving strategy.
+   * @param robotToCamera Transform3d from the center of the robot to the camera mount position (ie,
+   *     robot ➔ camera) in the <a href=
+   *     "https://docs.wpilib.org/en/stable/docs/software/advanced-controls/geometry/coordinate-systems.html#robot-coordinate-system">Robot
+   *     Coordinate System</a>.
+   * @return an {@link EstimatedRobotPose} with an estimated pose, timestamp, and targets used to
+   *     create the estimate.
+   */
+  public static Optional<EstimatedRobotPose> update(
+      List<PhotonTrackedTarget> targets,
+      double timestamp,
+      Matrix<N3, N3> cameraMatrix,
+      Matrix<N8, N1> distCoeffs,
+      PoseStrategy strat,
+      Transform3d robotToCamera,
+      AprilTagFieldLayout fieldTags,
+      Transform3d bestTF) {
+    Optional<EstimatedRobotPose> estimatedPose;
```

Comment:
I know virtually everything in this PR is just an import, but I'd prefer to see a `final` here to ensure we're not returning null accidentally.